### ISSUE TITLE
Website is slow on initial load (Component-preload.js modules ignored)

### ISSUE THREAD

**Issue by proehlen** [propellerlabsio/meteor-ui5-website#15]
Trying to use [this guide](http://pipetree.com/qmacro/blog/2015/07/speeding-up-your-ui5-app-with-a-component-preload-file/) to create a Component-preload.js file (see `component-preload` branch).

Having trouble producing any output from grunt task (see SAP/grunt-openui5#43 ).

**Comment by proehlen** [comment]
Subjectively seems faster now but, although it is loading the Component-preload.js file it is still loading other resources individually - e.g. Home.controller.js.

Requires further investigation.

**Comment by proehlen** [comment]
Files being fetched:
![component](https://cloud.githubusercontent.com/assets/8384068/18213963/e84cc0be-7174-11e6-80d5-fdcc539e1c56.png)
![component2](https://cloud.githubusercontent.com/assets/8384068/18213974/f695fc9e-7174-11e6-9ef6-eb1d60c701e7.png)

Exist in Component-preload.js:
![component3](https://cloud.githubusercontent.com/assets/8384068/18214025/2ab14a92-7175-11e6-99e4-73233db9810d.png)

**Comment by matz3** [comment]
I just checked your project and it seems that this is caused by the fact jQuery gets loaded after UI5 has been initialized already.
As UI5 brings it's own version of jQuery and extends it (`jQuery.sap` namespace) those modules are lost after your version of jQuery is loaded.

Unfortunately there is no error which gives a hint to this (but this should actually break more than just the preload).
You have to provide some configuration options to get the actual error logged. As the debug mode by default disables the preload files, they have to be activated again:
http://localhost:3000/?sap-ui-debug=true&sap-ui-xx-show-load-errors=true&sap-ui-xx-componentPreload=sync

This should log
`Uncaught TypeError: Cannot read property 'registerPreloadedModules' of undefined`

To fix this you should remove the jQuery dependency from your project or use the `sap-ui-core-nojQuery.js` bootstrap after loading your version of jQuery (see https://openui5.hana.ondemand.com/#docs/guide/91f1dd0c6f4d1014b6dd926db0e91070.html).

**Comment by proehlen** [comment]
@matz3 Really can't thank you enough for the above information - just amazingly helpful.  I've removed the meteor jquery package locally and it started loading the preload file.  The extra options with debug you've explained to me also helped me realize that the preload file wasn't being deployed to our Scalingo webserver because it was in the .gitignore file.  I've fixed that and it's now successfully loading from the web server.

Thanks again for your time and help!
